app/models/miq_task.rb

```diff
@@ -257,9 +257,8 @@ def task_results
     return miq_report_result.report_results unless miq_report_result.nil?
     unless binary_blob.nil?
       serializer_name = binary_blob.data_type
-      serializer_name = "Marshal" unless serializer_name == "YAML" # YAML or Marshal, for now
       serializer = serializer_name.constantize
-      result = serializer.load(binary_blob.binary)
+      result = serializer.respond_to?(:unsafe_load) ? serializer.unsafe_load(binary_blob.binary) : serializer.load(binary_blob.binary)
```

Comment:
Since this is _always_ a binary blob, can we just use `binary_blob.data`?

Reply:
Good eye.  The BinaryBlob#data commit was added after the fact but we can definitely eliminate the conditional here and delegate to `data` to do the `unsafe_load` vs `load`.

Reply:
Added cleanup PR #22866